### ISSUE TITLE
Device Not Mounted when Attached to WSL

### ISSUE THREAD

**Issue by jerryfleurival** [dorssel/usbipd-win#1065]
### _I see it created a directory under media folder called USB, But don't see drive directory of my USB device._

**_Should I see it in the list when I type command "lsblk" into bash shell?_**

**Comment by dorssel** [comment]
You should see it with `lsusb`. Did you enable USB storage drivers in the WSL kernel?

**Comment by dorssel** [comment]
This means it is not attached yet. Did you run `usbipd attach --wsl ...` ? What was the output?

**Comment by jerryfleurival** [comment]
I forgot about that. Now I see my device listed. But, I'm trying to still figure out how to go into the device.

Output:
![Image](https://github.com/user-attachments/assets/4d89fb5a-3361-4fc9-99cc-6cb9e20571ce)

**Comment by dorssel** [comment]
Do you have a WSL kernel with USB storage drivers enabled?

**Comment by jerryfleurival** [comment]
Yes, my WSL Kernel Version is 5.15.167.4-microsoft-standard-WSL2 and I am operating under Windows 10 
Version 10.0.19045 Build 19045.

Here is some Kernel Logs:
![Image](https://github.com/user-attachments/assets/12cd2036-ad51-43da-9f29-547d52b876c8)

----------------------------------------------------------------------
**Notes:**
Not listed when I type lsblk command.

![Image](https://github.com/user-attachments/assets/3bc4597d-cbed-4bb8-bdb5-22025cc92a5b)

WSL must not recognize it as a block device.

**Comment by dorssel** [comment]
Your kernel does *not* have a USB storage driver (the default WSL kernel doesn't have that). You either need to compile your own kernel (see https://github.com/dorssel/usbipd-win/wiki/WSL-support#building-your-own-wsl-2-kernel-with-additional-drivers) or use the pre-release version of WSL (Windows 11 only), which has it as a module.

**Comment by jerryfleurival** [comment]
_I am having issues with branching into linux-msft-wsl-5.15.167.4 and this might take awhile than I expect to figure out._

Output:

![Image](https://github.com/user-attachments/assets/a6e1a469-5b53-4664-80e4-0943ee0567fc)

**Comment by dorssel** [comment]
Yep, that is a known issue. You cannot checkout the Linux kernel on an NTFS (Windows) file system. The "aux" filename (extension or not) is special. See for example https://github.com/raspberrypi/linux/issues/1254. People have complained to both Microsoft and Linux ... nothing is done about it. You need to checkout in a VM, to a non-Windows mounted file system.

**Comment by bugparty** [comment]
Don't lose hope, I wrote an tutorial and it is not complex, just take 10 minutes without any sideeffects.
see here: https://www.linkedin.com/pulse/wsl2-sd-card-support-fix-guide-bowen-han-34x3c/

**Comment by jerryfleurival** [comment]
@dorssel Does this mean I have to reinstall usbipd in VM for my Ubuntu OS.

**Comment by dorssel** [comment]
No, `usbipd` is the server (host) side, which is Windows. You don't need to install anything in Ubuntu. You need to *replace* the WSL kernel, either with some pre-built kernel (as suggested above), or compile it yourself (also suggested above).

**Comment by jerryfleurival** [comment]
@dorssel **My current status:
           _ sdc1 is my USB storage device mounted on media folder._**


![Image](https://github.com/user-attachments/assets/fd39b3e6-aabe-4634-873a-be874bfcce34)


![Image](https://github.com/user-attachments/assets/59b95619-dbf5-4651-a545-99ab07712e4f)

I am using my other computer which has more resources to handle VM (VirtualBox) and OS is Windows 11.
I notice I have not use `attach `and `bind.`  I'm thinking that is for interacting with remote host or you can say working with two or more computer systems. Is that so?

**Comment by jerryfleurival** [comment]
@dorssel Okay. [Oracle VirtualBox](https://www.virtualbox.org/) gave me the GUI to Share and Attach USB device locally. So, I can confirm this  with `usbip` command like in my previous comment and be able to do other tasks like `usbipd`. 

 I have confirmed the use of `attach`,` detach`, and` bind` in VM to reach remote host.  Yes, for reaching the host, you first need `usbipd` running in Windows. I need --remote=<IPv4 address> --busid=<id> with of course admin powers to achieve this. It has automatically mounted inside Media directory. Now I can interact with the files and folders of my USB drive. 

Last, as you see in the screenshot above, I am using pre-built Kernel (Generic Kernel) under Ubuntu, which is fine I guess. But, since the PC I'm using is Windows 11, I could go along with using modules by following your steps in [WSL Support wiki](https://github.com/dorssel/usbipd-win/wiki/WSL-support#building-your-own-wsl-2-kernel-with-additional-drivers). Either way, I now know I have options. 

Let me know if I'm missing something or if something you like to add as a note. Thanks a lot!
